From 60e04048a0e2f3e33651c19e62861b41be4c290f Mon Sep 17 00:00:00 2001
From: Felix Fietkau <nbd@nbd.name>
Date: Fri, 17 Oct 2025 11:15:26 +0000
Subject: [PATCH] ubusd: fix ACL check for receiving events

Wildcard event match must also be subject to ACL checks

Reported-by: Karsten Sperling <ksperling@apple.com>
Signed-off-by: Felix Fietkau <nbd@nbd.name>
(cherry picked from commit 60e04048a0e2f3e33651c19e62861b41be4c290f)
[Backport to ubus 2022-02-21-b32a0e17 (OpenWrt 19.07, Gluon v2021.1.x) by
 Freifunk im Neanderland <projekt@neanderfunk.de>: code unchanged,
 hunks refreshed for line offsets only. Part of the CVE-2025-62526 fix
 set (d31effb, aa4a7ee, 60e0404), as in OpenWrt 24.10 a790196993.]
---
 ubusd_event.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ubusd_event.c b/ubusd_event.c
--- a/ubusd_event.c
+++ b/ubusd_event.c
@@ -90,7 +90,7 @@ static int ubusd_alloc_event_pattern(struct ubus_client *cl, struct blob_attr *m
 		len--;
 	}
 
-	if (pattern[0] && ubusd_acl_check(cl, pattern, NULL, UBUS_ACL_LISTEN))
+	if (ubusd_acl_check(cl, pattern, NULL, UBUS_ACL_LISTEN))
 		return UBUS_STATUS_PERMISSION_DENIED;
 
 	ev = calloc(1, sizeof(*ev) + len + 1);
